finality-aleph/src/validator_network/protocols/v0/mod.rs

```diff
@@ -217,24 +141,31 @@ mod tests {
         impl futures::Future<Output = Result<(), ProtocolError>>,
         impl futures::Future<Output = Result<(), ProtocolError>>,
         UnboundedReceiver<D>,
-        UnboundedReceiver<(AuthorityId, oneshot::Sender<()>)>,
-        UnboundedReceiver<(AuthorityId, Option<mpsc::UnboundedSender<D>>)>,
+        UnboundedReceiver<(
+            AuthorityId,
+            Option<mpsc::UnboundedSender<D>>,
+            ConnectionType,
+        )>,
+        UnboundedReceiver<(
+            AuthorityId,
+            Option<mpsc::UnboundedSender<D>>,
+            ConnectionType,
+        )>,
```

Comment:
just to support my other comment: these should be `UnboundedReceiver<ResultForService<D>>` (or maybe even onshots?)

Reply:
Partially fixed, partially see comment above.